torch_tvm/compiler.cpp

```diff
@@ -320,8 +342,22 @@ void TVMCompiler::run(Stack& stack) {
       value_to_ivalue[value] = optional_ivalue.value();
     }
     auto ivalue = value_to_ivalue.at(cache_[spec].input_values[i]);
-    auto tensor = ivalue.toTensor().to(at::kFloat);
-    auto dl_tensor = at::toDLPack(tensor);
+    //auto tensor = ivalue.toTensor().to(at::kFloat);
+    auto tensor = ivalue.toTensor();
+    DLManagedTensor* dl_tensor;
+    if (torch_tvm::utils::is_aligned(tensor.data_ptr(),
+          tvm::runtime::kAllocAlignment)) {
+      dl_tensor = at::toDLPack(tensor);
+    } else if (cache_[spec].tvm_param_tensors_.value_exists(value)) {
```

Comment:
Eventually we don't want to even compare with this value if we already know what's treated as weights in TVM as you probably will go though alter layout and have your own transformed weights.

Reply:
yes, but we cant couple this with whether weights get transformed. Transformation is optimization step and not always done or can be relied upon. Right? Unless I am missing something.

Reply:
Once you enable opt_lvl=3, you cannot expect that your weights are not changed.

Reply:
I dont think I am following you. The only expectation above condition sets is that we created and copied tensor that corresponds to some Value*. Here that tensor is weight. What happens inside tvm, as a result opt_lvl, is not what this intends to do.
Can you elaborate what you mean by "you cannot expect that your weights are not changed"? Are you talking about layout transformation of weights?

Reply:
> Are you talking about layout transformation of weights?

Yes.

Reply:
Only reason the check is done is because: at the time of relay graph creation, the only thing we have is Value*. But this is not strictly true. We create relay graph on the very first inference which means the weights are definitely available. I think we can have `std::pair<Value*, DLManagedTensorPtr>`, instead of `std::pair<Value*, bool>` that maps Value* to its mutability property. But it probably makes code little less readable. Thanks for the input.